### PR TITLE
Update SegFormer model card

### PR DESCRIPTION
**Pull request by GSNCodes**
# What does this PR do?

https://github.com/huggingface/transformers/issues/36979
Updates the model card of SegFormer to the standardized format for documentation.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/segformer.md [hunk 1]
```diff
@@ -1,113 +1,107 @@
...
+           <img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
+    </div>
+</div>
+
 # SegFormer
 
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-</div>
+[SegFormer: Simple and Efficient Design for Semantic Segmentation with Transformers](https://huggingface.co/papers/2105.15203) is a semantic segmentation model that combines a **hierarchical Transformer encoder** (Mix Transformer, MiT) with a **lightweight all-MLP decoder**. It avoids positional encodings and complex decoders, yet achieves **state-of-the-art performance** on benchmarks like ADE20K and Cityscapes while being efficient and robust.
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md]:
```suggestion
[SegFormer: Simple and Efficient Design for Semantic Segmentation with Transformers](https://huggingface.co/papers/2105.15203) is a semantic segmentation model that combines a hierarchical Transformer encoder (Mix Transformer, MiT) with a lightweight all-MLP decoder. It avoids positional encodings and complex decoders and achieves state-of-the-art performance on benchmarks like ADE20K and Cityscapes. This simple and lightweight design is more efficient and scalable.
```

File: docs/source/en/model_doc/segformer.md [hunk 2]
```diff
@@ -1,113 +1,107 @@
...
-| MiT-b0            | [2, 2, 2, 2]  | [32, 64, 160, 256]  | 256                     | 3.7            | 70.5                  |
-| MiT-b1            | [2, 2, 2, 2]  | [64, 128, 320, 512] | 256                     | 14.0           | 78.7                  |
-| MiT-b2            | [3, 4, 6, 3]  | [64, 128, 320, 512] | 768                     | 25.4           | 81.6                  |
-| MiT-b3            | [3, 4, 18, 3] | [64, 128, 320, 512] | 768                     | 45.2           | 83.1                  |
-| MiT-b4            | [3, 8, 27, 3] | [64, 128, 320, 512] | 768                     | 62.6           | 83.6                  |
-| MiT-b5            | [3, 6, 40, 3] | [64, 128, 320, 512] | 768                     | 82.0           | 83.8                  |
-
-Note that MiT in the above table refers to the Mix Transformer encoder backbone introduced in SegFormer. For
-SegFormer's results on the segmentation datasets like ADE20k, refer to the [paper](https://huggingface.co/papers/2105.15203).
+You can find all the original SegFormer checkpoints under the [Nvidia](https://huggingface.co/nvidia/models?search=segformer) organization.
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md]:
```suggestion
You can find all the original SegFormer checkpoints under the [NVIDIA](https://huggingface.co/nvidia/models?search=segformer) organization.
```

File: docs/source/en/model_doc/segformer.md [hunk 3]
```diff
@@ -1,113 +1,107 @@
...
 
-## Resources
+> [!TIP]
+> This model was contributed by [nielsr](https://huggingface.co/nielsr).
+>
+> Click on the SegFormer models in the right sidebar for more examples of how to apply SegFormer to different tasks like **semantic segmentation** and **image classification**.
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md]:
```suggestion
> Click on the SegFormer models in the right sidebar for more examples of how to apply SegFormer to different vision tasks.
```

File: docs/source/en/model_doc/segformer.md [hunk 4]
```diff
@@ -1,113 +1,107 @@
...
+
+The example below demonstrates how to run **semantic segmentation** with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md]:
```suggestion
The example below demonstrates semantic segmentation with [`Pipeline`] or the [`AutoModel`] class.
```

File: docs/source/en/model_doc/segformer.md [hunk 5]
```diff
@@ -1,113 +1,107 @@
...
-different layers, and thus combining both local attention and global attention to render powerful representations. We
-show that this simple and lightweight design is the key to efficient segmentation on Transformers. We scale our
-approach up to obtain a series of models from SegFormer-B0 to SegFormer-B5, reaching significantly better performance
-and efficiency than previous counterparts. For example, SegFormer-B4 achieves 50.3% mIoU on ADE20K with 64M parameters,
-being 5x smaller and 2.2% better than the previous best method. Our best model, SegFormer-B5, achieves 84.0% mIoU on
-Cityscapes validation set and shows excellent zero-shot robustness on Cityscapes-C.*
-
-The figure below illustrates the architecture of SegFormer. Taken from the [original paper](https://huggingface.co/papers/2105.15203).
-
-<img width="600" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/segformer_architecture.png"/>
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md:46]: Let's keep this image

File: docs/source/en/model_doc/segformer.md [hunk 6]
```diff
@@ -1,113 +1,107 @@
...
+> This model was contributed by [nielsr](https://huggingface.co/nielsr).
+>
+> Click on the SegFormer models in the right sidebar for more examples of how to apply SegFormer to different tasks like **semantic segmentation** and **image classification**.
+
+The example below demonstrates how to run **semantic segmentation** with [`Pipeline`] or the [`AutoModel`] class.
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
+
+```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md:43]:
```py
import torch
from transformers import pipeline

pipeline = pipeline(task="image-segmentation", model="nvidia/segformer-b0-finetuned-ade-512-512", torch_dtype=torch.float16)
pipeline("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg")
```

File: docs/source/en/model_doc/segformer.md [hunk 7]
```diff
@@ -1,113 +1,107 @@
...
+image = "your_image.png"
+outputs = segmenter(image)
+```
+
+</hfoption>
+<hfoption id="AutoModel">
 
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with SegFormer.
+```python
+from transformers import SegformerImageProcessor, SegformerForSemanticSegmentation
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md]:
```py
import requests
from PIL import Image
from transformers import AutoProcessor, AutoModelForSemanticSegmentation

url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
image = Image.open(requests.get(url, stream=True).raw)

processor = AutoProcessor.from_pretrained("nvidia/segformer-b0-finetuned-ade-512-512")
model = AutoModelForSemanticSegmentation.from_pretrained("nvidia/segformer-b0-finetuned-ade-512-512")

inputs = processor(images=image, return_tensors="pt")
outputs = model(**inputs)
logits = outputs.logits # shape [batch, num_labels, height, width]
```

File: docs/source/en/model_doc/segformer.md [hunk 8]
```diff
@@ -1,113 +1,107 @@
...
 
-- [`SegformerForSemanticSegmentation`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/semantic-segmentation).
-- A blog on fine-tuning SegFormer on a custom dataset can be found [here](https://huggingface.co/blog/fine-tune-segformer).
-- More demo notebooks on SegFormer (both inference + fine-tuning on a custom dataset) can be found [here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/SegFormer).
-- [Semantic segmentation task guide](../tasks/semantic_segmentation)
+</hfoption>
+
+</hfoptions>
+
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md]: The model isn't very big so no need to have a quantization example

File: docs/source/en/model_doc/segformer.md [hunk 9]
```diff
@@ -1,113 +1,107 @@
...
+quantization_config = BitsAndBytesConfig(load_in_8bit=True)
+model = AutoModelForSemanticSegmentation.from_pretrained(
+    "nvidia/segformer-b0-finetuned-ade-512-512",
+    quantization_config=quantization_config
+)
+```
+
+## Notes
+
+- SegFormer works with **any input size**, padding inputs to be divisible by `config.patch_sizes`.
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md:78]:
```suggestion
- SegFormer works with **any input size**, padding inputs to be divisible by `config.patch_sizes`.
- The most important preprocessing step is to randomly crop and pad all images to the same size (such as 512x512 or 640x640) and normalize afterwards.
```

File: docs/source/en/model_doc/segformer.md [hunk 10]
```diff
@@ -1,113 +1,107 @@
...
+- When preprocessing, be mindful of `do_reduce_labels`:
+  - Some datasets (like ADE20K) don’t include background in the labels → set `do_reduce_labels=True`.
+  - Other datasets do include background → set `do_reduce_labels=False`.
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md]:
```suggestion
- Some datasets (ADE20k) uses the `0` index in the annotated segmentation as the background, but doesn't include the "background" class in its labels. The `do_reduce_labels` argument in [`SegformerForImageProcessor`] is used to reduce all labels by `1`. To make sure no loss is computed for the background class, it replaces `0` in the annotated maps by `255`, which is the `ignore_index` of the loss function.

   Other datasets may include a background class and label though, in which case, `do_reduce_labels` should be `False`.
```

File: docs/source/en/model_doc/segformer.md [hunk 11]
```diff
@@ -1,113 +1,107 @@
...
+- Model variants differ in size and accuracy (MiT-B0 to MiT-B5). Example:
+  - **SegFormer-B4**: 50.3% mIoU on ADE20K with 64M parameters, 5x smaller and better than prior methods.
+  - **SegFormer-B5**: 84.0% mIoU on Cityscapes validation set, strong zero-shot robustness.
```
Comment (stevhliu) [inline docs/source/en/model_doc/segformer.md]:
```suggestion
```

**Review by stevhliu — COMMENTED** [review]
Nice work!

File: docs/source/en/model_doc/segformer.md (hunk 5 above)
Comment (GSNCodes) [reply docs/source/en/model_doc/segformer.md:46]: Added the image

File: docs/source/en/model_doc/segformer.md (hunk 8 above)
Comment (GSNCodes) [reply docs/source/en/model_doc/segformer.md]: Gotcha, removed it

**Comment by GSNCodes** [comment]
Thanks @stevhliu I've addressed all the comments :)

**Review by stevhliu — APPROVED** [review]
LGTM, thanks!
